libs/blocks/global-navigation/utilities/onImsReady.js

```diff
@@ -0,0 +1,31 @@
+let ready = false;
+let existingCall = null;
+const onInstance = 'onImsLibInstance';
+
+const onImsReady = (timeout = 3000) => {
+  existingCall = existingCall || new Promise((resolve, reject) => {
+    const onFail = () => {
+      if (!ready) reject();
+    };
+    const waitTimeout = setTimeout(onFail, timeout);
+    const onSuccess = (data) => {
+      const instance = data?.detail?.instance;
+      if (!instance) {
+        reject();
+        return;
+      }
+
+      ready = true;
+      window.removeEventListener(onInstance, onSuccess);
+      clearTimeout(waitTimeout);
+      resolve(instance);
+    };
+
+    window.addEventListener(onInstance, onSuccess);
+    window.dispatchEvent(new window.CustomEvent('getImsLibInstance'));
```

Comment:
What's the scenario where it's helpful to know when the logic starts looking for the instance?

Reply:
I was not aware that this is needed for IMS to return the instance, my previous comment is irrelevant.